R/library_call_linter.R

```diff
@@ -0,0 +1,60 @@
+#' Library call linter
+#'
+#' Force library calls to all be at the top of the script.
+#'
+#' @examples
+#'  # will produce lints
+#'  lint(
+#'    text = c("library(dplyr)", "print('test')", "library(tidyr)"),
+#'    linters = library_call_linter()
+#'  )
+#'
+#'  lint(
+#'    text = c("library(dplyr)", "print('test')", "library(tidyr)", "library(purrr)"),
+#'    linters = library_call_linter()
+#'  )
+#'
+#'  # okay
+#'  lint(
+#'    text = c("library(dplyr)", "print('test')"),
+#'    linters = library_call_linter()
+#'  )
+#'
+#'  lint(
+#'    text = c("# comment", "library(dplyr)"),
+#'    linters = library_call_linter()
+#'  )
+#'
+#' @evalRd rd_tags("library_call_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+library_call_linter <- function() {
+
+  xpath <- "
+  (//SYMBOL_FUNCTION_CALL[text() = 'library'])[last()]
+  //preceding::expr
```

Comment:
The // throws away the scope by restarting from the root element. Are you sure you don't want a single /?

Reply:
Thanks, yep, that was not my intention. I definitely misunderstood the XPATH documentation here.  I'll make the update to / here and for line 36 and 37.